### ISSUE TITLE
Unable to paste magnet links

### ISSUE THREAD

**Issue by anohren** [transmission-remote-gui/transgui#1193]
1. Operating system
macOS 10.14.1
2. Transmission Deamon
Not relevant
3. Transmission Remote GUI
5.16.0

In the "Add torrent link" dialog I cannot right click to bring up a context menu in order to paste a link from clipboad, and Cmd + V results in a dialog "access violation" followed by a crash.

**Comment by montykabinski** [comment]
I have the exact same problem with 5.16.0 on Ubuntu 18.10 ( 4.18.0-15-generic #16-Ubuntu SMP Thu Feb 7 10:56:39 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux).

Doesn't always give an access violation, but does about 80% of the time.

**Comment by leonsoft-kras** [comment]
We need a crash report, but for some reason we still have not implemented it.

**Comment by leonsoft-kras** [comment]
@PeterDaveHello, what are your suggestions?

**Comment by antekgla** [comment]
I dont have access to a MacOS or Linux machine.... 😢 
Only can test in Windows (7 and 10)

.... I can't reproduce this error in 5.16 in Windows.

**Comment by leonsoft-kras** [comment]
It is necessary to make a record of crash reports for all operating systems. 
Unlock my revision and not reinvent the wheel.
Now compilation flags dont allow generating a file with crash report data.

**Comment by anohren** [comment]
If it helps it seems that the clipboard is abused by the app when the option to automatically add torrents from clipboad is enabled; whenever the app gains focus it reads the clipboard, and if it contains anything with "magnet:" it goes ahead and erases the clipboard for no apparenty reason. 

Since the app overwrites the clipboard without being asked to, it generally makes it very hard to understand what's happening while trying to interact with the app using the clipboard.

**Comment by antekgla** [comment]
>  it goes ahead and erases the clipboard for no apparent reason.

The reason why the clipboard is erased is because once the magnet is added if the clipboard still contains the magnet link the app would be try again to add it. So the original author (5.0.1) decide what once the link is added the clipboard is erased to avoid such thing.

**Comment by anohren** [comment]
Thanks for the clarification!

Between the two options to either erase user data without permission or to silently ignore a duplicate torrent I'd have to say that the latter is preferable.

I don't know if I manually enabled this option in the past and then forgot about it, or if it was enabled by default (which would be worse), but in either case I expect my clipboad to be left alone unless I click a red button titled "Erase clipboard".

No wonder I've never been able to paste magnet links into the text field, wondering if I didn't copy it correctly, only to exit the dialog and discover that the torrent has magically started. 

It doesn't explain the access violation though.

**Comment by antekgla** [comment]
This ALWAYS was the behavior (at least from the 5.0.1 version what is the base for the @leonsoft-kras version).
If you want to copy the link for yourself in the Add Magnet form you should un-check the option:

`Automatically add torrent links from the clipboard`

in Application Options.
With this option unchecked your clipboard NEVER would be erased again by TransGUI.

**Comment by anohren** [comment]
> This ALWAYS was the behavior (at least from the 5.0.1 version what is the base for the @leonsoft-kras version).

Ok, it has always deleted user data without confirmation. I'm afraid I don't see your point though. 



>  If you want to copy the link for yourself in the Add Magnet form you should un-check the option:
> 
> `Automatically add torrent links from the clipboard`
> 
> in Application Options.
> With this option unchecked your clipboard NEVER would be erased again by TransGUI.

Yes, obviously, but that's beside the point, since I'm only talking about the behavior when the option is enabled, not when it's disabled.

**Comment by antekgla** [comment]
> Yes, obviously, but that's beside the point, since I'm only talking about the behavior when the option is enabled, not when it's disabled

What is the point of use the Add Torrent/Magnet option when this is done automatically?
My opinion is what when you go to Add Torrent/Magnet form the link you previously copy is not there because the link is already added and the clipboard erased like I explain before.

**Comment by anohren** [comment]
> What is the point of use the Add Torrent/Magnet option when this is done automatically?

I have never argued that there is a point to that. I have only alluded to the fact that I was completely unaware that the option to do it automatically from the clipboard was enabled, _let alone existed_. That is enough to explain why I tried to paste my magnet link manually.



>  My opinion is what when you go to Add Torrent/Magnet form the link you previously copy is not there because the link is already added and the clipboard erased like I explain before.

Yes, I also concluded that a few posts ago:

> No wonder I've never been able to paste magnet links into the text field [...]

I'm still unsure as to what you're trying to say. Are you arguing that it's ok to delete the user's data they've put in their system-wide clipboard?

**Comment by antekgla** [comment]
> I'm still unsure as to what you're trying to say. Are you arguing that it's ok to delete the user's data that they've put into their system-wide clipboard?

I dont know if is wright or wrong to delete the clipboard. It just a way to do it as I explain messages ago.

TransGUI dont delete all content of your clipboard... ONLY when are links to magnets or torrents and only when the app has focus. TransGUI is not all the time checking your clipboard and erase the content. If you want your link to magnet in the clipboard for other reason what adding to TransGUI simply dont give focus to TransGUI. The link remains there if the app dont have focus.

I do not see it as serious.

If you dont want this behavior... disable the option to automatically add torrent/magnets links.

**Comment by anohren** [comment]
> I dont know if is right or wrong to delete the clipboard. It just a way to do it as I explain messages ago.

Yes, thank you again for your explanation. I must have mistakenly got the impression that you had strong feelings about this issue when you appeared to start shouting about "always" and "never".

> If you want your link to magnet in the clipboard for other reason what adding to TransGUI simply dont give focus to TransGUI. The link remains there if the app dont have focus.

This is entirely orthogonal to the problem, since I'm talking about the scenario when it _does_ get focus and _does_ get erased. Talking about a scenario where the problem doesn't occur as though it were a solution doesn't really move us forward. If you personally don't feel inclined to fix it, that's fine of course.

> TransGUI dont delete all content of your clipboard... ONLY when are links to magnets or torrents and only when the app has focus. TransGUI is not all the time checking your clipboard and erase the content.

Well, yes, obviously. This also has nothing to do with the problem that I'm talking about. I'm talking about what it does do, not all the things that it doesn't do. I'm hoping that we can focus on the part of the problem that exists.

> If you dont want this behavior... disable the option to automatically add torrent/magnets links

You still seem to miss the point of reporting a problem. "If you find a problem with a feature when it's enabled, just disable it" is like saying "If you find a bug in the app, just don't use the app" -- this is just not how I view bug reporting, since that would make it pointless to report issues at all. I'm only trying to give a description and rationale about a problem -- not put any sort of priority on it. 

The issue is there, waiting -- whether the feature is enabled or not -- as long as the feature can be enabled. I can't really change that, and existing workarounds don't change it either.

**Comment by Pentaphon** [comment]
Interesting. I wonder if its Mac related.
